### PR TITLE
"In-memory" repos are bare by default

### PR DESCRIPTION
**Pull request by ethomson**
A repository created using `git_repository_new` lacks a working directory and thus should report `true` when queried with `git_repository_is_bare`.

(This repository will no longer be bare when `set_workdir` is called, if it is called.)

### CODE REVIEW

**Comment by nulltoken** [comment]
How about a tiny test (including an eventual call to `set_workdir`)?

**Comment by ethomson** [comment]
There ya go.

File: tests/repo/new.c [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+	cl_git_pass(git_repository_new(&repo));
+	cl_assert_equal_b(true, git_repository_is_bare(repo));
+
+	cl_git_pass(git_repository_set_workdir(repo, clar_sandbox_path(), 0));
+	cl_assert_equal_b(false, git_repository_is_bare(repo));
+
+	git_repository_free(repo);
+}
+
```
Comment (nulltoken) [inline tests/repo/new.c]: Oh! Look! An additional trailing newline! :trollface:

**Comment by nulltoken** [comment]
:+1:

File: tests/repo/new.c (hunk 1 above)
Comment (carlosmn) [inline tests/repo/new.c]: Better safe than sorry

**Pull request merged**